caringcaribou/utils/iso14229_1.py

```diff
@@ -293,19 +295,26 @@ def receive_response(self, wait_window):
         return response
 
     @staticmethod
-    def is_positive_response(response):
+    def is_positive_response(response: List[int]) -> bool:
         """
         Returns a bool indicating whether 'response' is positive
 
         :param response: ISO-14229-1 response data
-        :return: False if response is a NEGATIVE_RESPONSE,
-                 True otherwise
-        """
-        if (response is not None and
-           len(response) > 0 and
-           response[0] != Constants.NR_SI):
-            return True
-        return False
+        :return: False if response is a NEGATIVE_RESPONSE, True otherwise
+        :rtype: bool
+        """
+        return response is not None and len(response) >= 2 and response[0] != Constants.NR_SI
+
+    @staticmethod
+    def is_negative_response(response: List[int]) -> bool:
+        """
+        Returns a bool indicating whether 'response' is negative
+
+        :param response: ISO-14229-1 response data
+        :return: True if response is a NEGATIVE_RESPONSE, False otherwise
+        :rtype: bool
+        """
+        return response is not None and len(response) >= 3 and response[0] == Constants.NR_SI
```

Comment:
I believe the length of a negative response should be *exactly* 3 bytes (meaning that messages longer than 3 bytes would be invalid). Have you seen any standard case or implementation which differs from this?

Reply:
Hmm yes, good point, i havent noticed other length of negative response 👍